c7n/resources/ebs.py

```diff
@@ -446,6 +449,57 @@ def process(self, resources, event=None):
         return [r for r in resources if r['VolumeId'] in flagged]
 
 
+@filters.register('health-event')
+class HealthFilter(HealthEventFilter):
+
+    def process(self, resources, event=None):
+        if not resources:
+            return resources
+
+        client = local_session(self.manager.session_factory).client('health')
+        paginator = client.get_paginator('describe_events')
+        m = self.manager
+        result_map = {}
+
+        f = {'services': [m.data['resource'].upper()],
```

Comment:
this implementation seems like a lot of code duplication from the base class, i'd suggest refactoring the base class so you can get the custom logic for lost volume without the duplication.